event_sourcery_sqlalchemy/subscription.py

```diff
@@ -30,4 +54,90 @@ def subscribe_to_events(
         timelimit: timedelta,
         events: list[str],
     ) -> Iterator[list[RecordedRaw]]:
-        raise NotImplementedError
+        return GapDetectingIterator(
+            session=self._session,
+            gap_retry_interval=self.GAP_RETRY_INTERVAL,
+            start_from=start_from,
+            batch_size=batch_size,
+            timelimit=timelimit,
+            filtering=models.Event.name.in_(events),
+        )
+
+
+@dataclass
+class Cursor:
+    position: Position
+
+    def advance(self, batch: list[models.Event]) -> None:
+        if len(batch) > 0:
+            self.position = batch[-1].id
+
+
+class GapDetectingIterator(Iterator[list[RecordedRaw]]):
+    def __init__(
+        self,
+        session: Session,
+        gap_retry_interval: timedelta,
+        start_from: Position,
+        batch_size: int,
+        timelimit: timedelta,
+        filtering: Any | None = None,  # TODO: _ColumnExpressionArgument[bool]
```

Comment:
Just import this private type.

Reply:
Or remove it. 

Declare
```python
GetBatch: TypeAlias = Callable[[Position, int], list[models.Event]]
```

and pass this instead of session, filtering, and managing query configuration here.

Reply:
Done, I changed signature to `(Position) -> list[models.Event]`